add-on/panels/js/copy.jsx

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global Components */
+let { utils: Cu } = Components;
+
+var loop = loop || {};
+loop.copy = (mozL10n => {
+  "use strict";
+
+  /**
+   * Copy panel view.
+   */
+  let CopyView = React.createClass({
+    getInitialState() {
+      return { checked: false };
+    },
+
+    _dispatch(accept) {
+      window.dispatchEvent(new CustomEvent("CopyPanelClick", {
+        detail: {
+          accept,
+          stop: this.state.checked
+        }
+      }));
+    },
+
+    handleAccept() {
+      this._dispatch(true);
+    },
+
+    handleCancel() {
+      this._dispatch(false);
+    },
+
+    handleToggle() {
+      this.setState({ checked: !this.state.checked });
+    },
+
+    render() {
+      return (
+        <div className="copy-content">
+          <div className="copy-body">
+            <img className="copy-logo" src="shared/img/helloicon.svg" />
+            <h1 className="copy-message">
+              {mozL10n.get("copy_panel_message")}
+              <label className="copy-toggle-label">
+                <input onChange={this.handleToggle} type="checkbox" />
+                {mozL10n.get("copy_panel_dont_show_again_label")}
+              </label>
+            </h1>
+          </div>
+          <button className="copy-button" onClick={this.handleCancel}>
+            {mozL10n.get("copy_panel_cancel_button_label")}
+          </button>
+          <button className="copy-button" onClick={this.handleAccept}>
+            {mozL10n.get("copy_panel_accept_button_label")}
+          </button>
+        </div>
+      );
+    }
+  });
+
+  /**
+   * Copy panel initialization.
+   */
+  function init() {
+    let { LoopAPI } = Cu.import("chrome://loop/content/modules/MozLoopAPI.jsm", {});
+    let requests = ["GetAllStrings", "GetLocale", "GetPluralRule"];
+    return Promise.all(requests.map(name => new Promise(resolve => {
```

Comment:
Splitting out a getFromLoopAPI function or something like that could be another possibility.